src/ConfigFetcher.ts

```diff
@@ -0,0 +1,45 @@
+import { IConfigFetcher, OptionsBase, FetchResult, ProjectConfig } from "configcat-common";
+
+export class HttpConfigFetcher implements IConfigFetcher {
+    fetchLogic(options: OptionsBase, lastEtag: string, callback: (result: FetchResult) => void): void {
+        const fetchId = Math.round(Math.random() * 100000);
```

Comment:
This solution looks really dangerous to me. As far as I know, the suggested solution for this is an AbortController.
https://dmitripavlutin.com/timeout-fetch-request/
https://stackoverflow.com/questions/46946380/fetch-api-request-timeout

Maybe polyfills should be added for it.

I'm not sure if this works with manifest v3. If not, please try to find a much safer solution than generating a random number.